tests/test_bounce_integral.py

```diff
@@ -0,0 +1,739 @@
+        #  below, but should everything related to B be normalized?
+        #  or just things relavant for computing bounce points?
+        #  e.g. should I normalize B dot e^zeta = B^zeta by Bref as well?
+        eq,
+        rho,
+        alpha,
+        knots=zeta,
+        check=True,
+        normalize=Bref,
+    )
+    data_keys = [
+        "|grad(psi)|^2",
+        "grad(psi)",
+        "B",
+        "iota",
+        "|B|",
+        "B^zeta",
+        "cvdrift0",
+        "cvdrift",
+        "gbdrift",
+    ]
+    # FIXME (outside scope of the bounce branch):
+    #  override_grid should not be required for the test to pass.
+    #  and anytime override_grid is true we should print a blue warning.
+    data_bounce = eq.compute(data_keys, grid=items["grid_desc"], override_grid=False)
+
+    # normalizations
+    bmag = data_bounce["|B|"] / Bref
+    B0 = np.mean(bmag)
+    bmag_an = B0 * (1 - epsilon * np.cos(theta_PEST))
+    np.testing.assert_allclose(bmag, bmag_an, atol=5e-3, rtol=5e-3)
+
+    x = Lref * rho
+    s_hat = -x / iota * shear / Lref
+    gradpar = Lref * data_bounce["B^zeta"] / data_bounce["|B|"]
+    gradpar_an = 2 * Lref * data_bounce["iota"] * (1 - epsilon * np.cos(theta_PEST))
+    np.testing.assert_allclose(gradpar, gradpar_an, atol=9e-3, rtol=5e-3)
+
+    # Comparing coefficient calculation here with coefficients from compute/_metric
+    cvdrift = -2 * np.sign(psi_boundary) * Bref * Lref**2 * rho * data_bounce["cvdrift"]
+    gbdrift = -2 * np.sign(psi_boundary) * Bref * Lref**2 * rho * data_bounce["gbdrift"]
+    dPdrho = np.mean(-0.5 * (cvdrift - gbdrift) * data_bounce["|B|"] ** 2)
+    alpha_MHD = -dPdrho * 1 / data_bounce["iota"] ** 2 * 0.5
+
+    gds21 = (
+        -np.sign(iota)
+        * dot(data_bounce["grad(psi)"], data_bounce["grad(alpha)"])
+        * s_hat
+        / Bref
+    )
+    fudge_factor2 = 0.19
+    gbdrift_an = fudge_factor2 * (
+        -s_hat + (np.cos(theta_PEST) - gds21 / s_hat * np.sin(theta_PEST))
+    )
+
+    fudge_factor3 = 0.07
+    cvdrift_an = gbdrift_an + fudge_factor3 * alpha_MHD / bmag**2
+    # Comparing coefficients with their analytical expressions
+    np.testing.assert_allclose(gbdrift, gbdrift_an, atol=1.5e-2, rtol=5e-3)
+    np.testing.assert_allclose(cvdrift, cvdrift_an, atol=1.8e-2, rtol=5e-3)
+
+    # Values of pitch angle lambda for which to evaluate the bounce averages.
+    pitch = np.linspace(1 / np.max(bmag), 1 / np.min(bmag), 11)
+    pitch = pitch.reshape(pitch.shape[0], -1)
+
+    k2 = 0.5 * ((1 - pitch * B0) / epsilon + 1)
+    # Fixme: What exactly is this a function of?
+    #  cvdrift, gbdrift is a grid quantity, so grid.num_nodes length
+    #  on a single field line grid -> so it has length number of zeta points
+    #  So bavg_drift_an has shape shape (number of pitch, number of zeta points).
+    #  For a fixed pitch at index i, what is difference bavg_drift_an[i, j]
+    #  and bavg_drift_an[i, j+1]?
+    bavg_drift_an = (
```

Comment:
@rahulgaur104 Could you help me interpret this? Thank you!

Reply:
Yeah, I'll do it tonight. You are right the bounce-averaged drift should only be a function of the pitch angle so basically these expressions should just be of the form scalar_1 * ellipe(k) + scalar_2 * ellipk(k) where k is a function of the pitch angle.

Reply:
I am trying that right now but it requires some tedious algebra. Also, did you change the definition of epsilon in this test?

Reply:
No, earlier epsilon was defined the same in the code as Lref * np.sqrt(psi) and I changed it to Lref * rho when refactoring